packages/realm/src/Configuration.ts

```diff
@@ -258,11 +276,12 @@ export function validateObjectSchema(
     }
     assert.object(clazz.schema, "schema static");
     validateObjectSchema(clazz.schema);
-  } else {
-    // Schema is passed as an object
-    assert.object(objectSchema, "the object schema", false);
+  }
+  // Schema is passed as an object (ObjectSchema)
+  else {
+    assert.object(objectSchema, "object schema", false);
     const { name: objectName, properties, primaryKey, asymmetric, embedded } = objectSchema;
-    assert.string(objectName, "the object schema name");
+    assert.string(objectName, "name (the object schema name)");
     assert.object(properties, `${objectName}.properties`, false);
```

Comment:
Just a suggestion - I think using the `.` here could cause some confusion.

```suggestion
    assert.object(properties, `properties on ${objectName}`, false);
```

Reply:
I'll go over the places where I use `.` and update it. E.g. see `Configuration.ts` where we validate the user input, I've used `"TheObjectName.thePropertyName"` but I see the confusion with how it looks like static types.

`#` was used before, can this be an option now? A good thing about being able to provide just one single "word" to the `assert` methods is due to the formatting of the error message from `TypeAssertionError.message()`.

Passing e.g. `"Person#age"` formats to `.. 'Person#age' ..`.
Passing e.g. `"age on Person"` formats to `.. 'age on Person' ..`.

We could have `TypeAssertionError.message()` accept a param to conditionally surround the value in single quotes, but then each of the methods on `assert` would also need this param.. Don't know how worth it it would be in order to instead be able to show e.g. `.. 'age' on 'Person' ..`